### PR TITLE
Set bigquery connection import id correctly.

### PR DESCRIPTION
**Pull request by ndmckinley**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/7615.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
bigqueryconnection: fixed failure to import a resource if it has a non-default project or location.
```

### CODE REVIEW

File: products/bigqueryconnection/terraform.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       cloudSql.credential.password: !ruby/object:Overrides::Terraform::PropertyOverride
         sensitive: true
     id_format: "{{name}}"
-    import_format: ["{{name}}"]
+    import_format: ["projects/{{project}}/locations/{{location}}/connections/{{connection_id}}", "{{project}}/{{location}}/{{connection_id}}", "{{location}}/{{connection_id}}"]
```
Comment (melinath) [inline products/bigqueryconnection/terraform.yaml:24]: This looks about right based on other `import_format`s.

Should `id_format` also be updated? It looks like the id_format is sometimes the long form and sometimes `{{name}}` so I'm not clear on whether that's intentional or legacy.

Is there a way to add a test to ensure that this works properly?

Comment (ndmckinley) [reply products/bigqueryconnection/terraform.yaml:24]: The `id_format` is set after apply, so it's safe to use `name`, which is an output field.  That's usually the difference - sometimes `name` is an output and sometimes it's an input.  When it's an output, it should be the ID, when it's an input, ID format should be the long form.  Not to say we haven't made occasional mistakes along the way, but that *should* be the difference, at least.

Comment (ndmckinley) [reply products/bigqueryconnection/terraform.yaml:24]: A test... well, there should be, right?  But the thing is, the existing test should have caught this.  I'm going to spend a little extra time trying to figure out why it didn't.

Comment (ndmckinley) [reply products/bigqueryconnection/terraform.yaml:24]: ah.  we don't actually do import state verify in the tests!  that's why!  Let me fix that.

Comment (ndmckinley) [reply products/bigqueryconnection/terraform.yaml:24]: Okay!  Finally done.  That was a tough one but I got it once I got started, haha.

**Comment by melinath** [comment]
@ndmckinley thanks for adding the tests! It looks like there were some test failures, but they seem unrelated; rerunning gcb to see if they were intermittent.

**Review by melinath — APPROVED** [review]
LGTM

**Pull request merged**